lib/commands/connect/shared.js

```diff
@@ -109,77 +110,69 @@ let request = exports.request = function(token, method, path, data){
   return promise;
 };
 
-let withUserConnections = exports.withUserConnections = function(token, appName, flags, allowNone, heroku){
+let withUserConnections = exports.withUserConnections = co.wrap(function* (token, appName, flags, allowNone, heroku){
   global_config();
 
   let resource_id = null;
   if (flags && flags.resource) {
     console.log("Calling Platform API for resouce id");
-    return Q.Promise(function(resolve, reject) {
-      return heroku.apps(appName).addons(flags.resource).info(function(err, addon) {
-        if(err) {
-          if (err.statusCode === 404) {
-            console.log("Connection with resource name '" + flags.resource + "' not found");
-            reject();
-          } else {
-            console.log(err);
-            reject(err);
-          }
+
+    return co(function* () {
+      try {
+        let addon = yield heroku.apps(appName).addons(flags.resource).info();
+      } catch(err) {
+        if (err.statusCode === 404) {
+          return yield Promise.reject("Connection with resource name '" + flags.resource + "' not found");
         } else {
-          flags.resource_id = addon.id;
-          delete flags['resource'];
-          withUserConnections(token, appName, flags, allowNone, heroku).then(resolve);
+          return yield Promise.reject(err);
         }
-      });
-
+      }
+      flags.resource_id = addon.id;
+      delete flags['resource'];
+      return yield Promise.resolve(withUserConnections(token, appName, flags, allowNone, heroku));
```

Comment:
Yeah, that's pretty much what I figured. But I wasn't sure because previously, `resolve()` on its own seemed to provide the promise equivalent of a return value, and I expected `yield Promise.resolve()` to do the same, but it really doesn't. I have a feeling I'm still missing some core piece of knowledge here, but at last I know what works. For now, that's good enough. :smile: